controllers/quoteattestation_controller.go

```diff
@@ -154,91 +155,61 @@ func (r *QuoteAttestationReconciler) Reconcile(ctx context.Context, req ctrl.Req
+	l.Info("Using provisioned secrets")
+	for _, signerName := range attestReq.Spec.SignerNames {
+		secret, ok := attestReq.Status.Secrets[signerName]
+		if !ok {
+			gotAllSecrets = false
+			l.Info("Secret not ready", "for signer", signerName)
+			continue
 		}
-	} else if secretsReady.Status == v1.ConditionFalse && secretsReady.Reason != v1alpha1.ReasonTCSReconcile {
-		// Secret preperation failure at attestation-controller side
-		l.V(3).Info("CA secret failure", "reason", secretsReady.Reason, "message", secretsReady.Message)
-		setSignerFailure(secretsReady)
-		return ctrl.Result{}, nil
-	} else {
-		gotAllSecrets := true
-		// attestation passed. Quote get verified
-		l.Info("Using provisioned secrets")
-		for _, signerName := range attestReq.Spec.SignerNames {
-			secret, ok := attestReq.Status.Secrets[signerName]
-			if !ok {
-				gotAllSecrets = false
-				l.Info("Secret not ready", "for signer", signerName)
-				continue
-			}
-			var provisionError error
-			if secret.SecretType == KMRABased {
-				l.Info("Using KMRA based secret.", "secretName", secret.SecretName)
-				provisionError = r.loadSecret(ctx, signerName, secret.SecretName, req.Namespace)
-			} else {
-				provisionError = fmt.Errorf("unsupported secret type: %v", secret.SecretType)
-			}
-			if provisionError != nil {
-				l.Info("CA provisioning", "error", provisionError)
-				s, _ := r.KeyProvider.GetSignerForName(signerName)
-				s.SetError(provisionError)
-				reqCopy := attestReq.DeepCopy()
-				attestReq.Status.SetCondition(v1alpha1.ConditionCASecretReady, v1.ConditionFalse, v1alpha1.ReasonTCSReconcile, provisionError.Error())
-				if err := r.Status().Patch(context.TODO(), &attestReq, client.MergeFrom(reqCopy)); err != nil {
-					r.Log.V(3).Info("Failed to update attestation status", "error", err)
-				}
-				return retry, nil
-			}
+		var provisionError error
+		if secret.SecretType == KMRABased {
+			l.Info("Using KMRA based secret.", "secretName", secret.SecretName)
+			provisionError = r.loadSecret(ctx, signerName, secret.SecretName, req.Namespace)
+		} else {
+			provisionError = fmt.Errorf("unsupported secret type: %v", secret.SecretType)
 		}
-		if gotAllSecrets {
-			r.done()
-			l.V(1).Info("Attestation passed. Private key(s) saved to enclave")
-			reqCopy := attestReq.DeepCopy()
-			attestReq.Status.SetCondition(v1alpha1.ConditionReady, v1.ConditionTrue, v1alpha1.ReasonTCSReconcile, "All CA keys and certificates stored in Enclace.")
-			if err := r.Status().Patch(context.TODO(), &attestReq, client.MergeFrom(reqCopy)); err != nil {
-				r.Log.V(3).Info("Failed to update attestation status", "error", err)
-			}
+		if provisionError != nil {
+			l.Info("CA provisioning", "error", provisionError)
+			s, _ := r.KeyProvider.GetSignerForName(signerName)
+			s.SetError(provisionError)
+			return retry, nil
 		}
-		return retry, nil
+	}
+	if gotAllSecrets {
+		r.done()
+		l.V(1).Info("Attestation passed. Private key(s) saved to enclave")
```

Comment:
Are the keys saved to the enclave at this point?

Reply:
Yes.